src/core/request/engines/composition/test/engine.spec.ts

```diff
@@ -0,0 +1,418 @@
+import Async from 'core/async';
+import request, { RequestError } from 'core/request';
+import { compositionEngine } from 'core/request/engines/composition';
+import { createServer } from 'core/request/engines/composition/test/server';
+
+// eslint-disable-next-line max-lines-per-function
+describe('core/request/engines/composition with request', () => {
+	let server: ReturnType<typeof createServer>;
+
+	beforeAll(() => {
+		server = createServer();
+	});
+
+	beforeEach(() => {
+		server.clearHandles();
+		server.handles.json1.response(200, {test: 1});
+	});
+
+	afterAll(() => {
+		server.destroy();
+	});
+
+	it('engine destructor call should lead to triggering the destructors of all the providers created by the engine', async () => {
+		try {
+			let r;
+
+			const engine = compositionEngine([
+				{
+					request: () => r = request('http://localhost:5000/json/1'),
+					as: 'result'
+				}
+			]);
+
+			await request('', {engine}).data;
+
+			const
+				requestResponseObject = await r,
+				spy = jest.spyOn(requestResponseObject, 'destroy');
+
+			expect(spy).toHaveBeenCalledTimes(0);
+			engine.destroy();
+
+			expect(spy).toHaveBeenCalledTimes(1);
+		} catch (err) {
+			console.log('error engine spec');
+			console.log(err);
+		}
+
+	});
+
+	it('should trigger a call to dropCache on all providers that were created by the engine when dropCache is called on the engine', async () => {
+		let r;
+
+		const engine = compositionEngine([
+			{
+				request: () => r = request('http://localhost:5000/json/1'),
+				as: 'result'
+			}
+		]);
+
+		await request('', {engine}).data;
+
+		const
+			requestResponseObject = await r,
+			spy = jest.spyOn(requestResponseObject, 'dropCache');
+
+		expect(spy).toHaveBeenCalledTimes(0);
+		engine.dropCache();
+
+		expect(spy).toHaveBeenCalledTimes(1);
+	});
+
+	it('error in a request without failCompositionOnError should not throw an error.', async () => {
```

Comment:
Может?
```suggestion
	it('should not throw an error in a request without failCompositionOnError.', async () => {
```

И ниже тоже